crates/bevy_core/src/time/timer.rs

```diff
@@ -10,227 +11,455 @@ use bevy_utils::Duration;
-    }
-
-    #[inline]
-    pub fn unpause(&mut self) {
-        self.paused = false
+    pub fn finished(&self) -> bool {
+        self.finished
     }
 
+    /// Returns `true` only on the tick the timer reached its duration.
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// use std::time::Duration;
+    /// let mut timer = Timer::from_seconds(1.0, false);
+    /// timer.tick(Duration::from_secs_f32(1.5));
+    /// assert!(timer.just_finished());
+    /// timer.tick(Duration::from_secs_f32(0.5));
+    /// assert!(!timer.just_finished());
+    /// ```
     #[inline]
-    pub fn paused(&self) -> bool {
-        self.paused
+    pub fn just_finished(&self) -> bool {
+        self.times_finished > 0
     }
 
     /// Returns the time elapsed on the timer. Guaranteed to be between 0.0 and `duration`.
     /// Will only equal `duration` when the timer is finished and non repeating.
+    ///
+    /// See also [`Stopwatch::elapsed`](Stopwatch::elapsed).
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// use std::time::Duration;
+    /// let mut timer = Timer::from_seconds(1.0, false);
+    /// timer.tick(Duration::from_secs_f32(0.5));
+    /// assert_eq!(timer.elapsed(), Duration::from_secs_f32(0.5));
+    /// ```
     #[inline]
-    pub fn elapsed(&self) -> f32 {
-        self.elapsed
-    }
-
-    #[inline]
-    pub fn set_elapsed(&mut self, elapsed: f32) {
-        self.elapsed = elapsed
-    }
-
-    #[inline]
-    pub fn duration(&self) -> f32 {
-        self.duration
+    pub fn elapsed(&self) -> Duration {
+        self.stopwatch.elapsed()
     }
 
+    /// Returns the time elapsed on the timer as a `f32`.
+    /// See also [`Timer::elapsed`](Timer::elapsed).
     #[inline]
-    pub fn set_duration(&mut self, duration: f32) {
-        self.duration = duration
+    pub fn elapsed_secs(&self) -> f32 {
+        self.stopwatch.elapsed_secs()
     }
 
-    /// Returns the finished state of the timer.
+    /// Sets the elapsed time of the timer without any other considerations.
     ///
-    /// Non-repeating timers will stop tracking and stay in the finished state until reset.
-    /// Repeating timers will only be in the finished state on each tick `duration` is reached or exceeded, so in that case
-    /// this function is equivalent to `just_finished`.
+    /// See also [`Stopwatch::set`](Stopwatch::set).
+    ///
+    /// #
+    /// ```
+    /// # use bevy_core::*;
+    /// use std::time::Duration;
+    /// let mut timer = Timer::from_seconds(1.0, false);
+    /// timer.set_elapsed(Duration::from_secs(2));
+    /// assert_eq!(timer.elapsed(), Duration::from_secs(2));
+    /// // the timer is not finished even if the elapsed time is greater than the duration.
+    /// assert!(!timer.finished());
+    /// ```
+    /// ```
     #[inline]
-    pub fn finished(&self) -> bool {
-        self.finished
+    pub fn set_elapsed(&mut self, time: Duration) {
+        self.stopwatch.set(time);
```

Comment:
These method names should probably be consistent. I think `set_elapsed` is clearer